src/Core/GitEventsMonitor.cs

```diff
@@ -65,7 +65,7 @@ private void WatchGitEvents(string repoFolder, IFileSystemWatcherFactory fileSys
 
             fileSystemWatcher.Filter = HEADFile;
             
-            fileSystemWatcher.Changed += HeadFileChanged;
+            fileSystemWatcher.Renamed += HeadFileChanged;
```

Comment:
Why are we only looking at file-rename events?

Does this have anything to do with the main purpose of this PR, or is it a completely separate bug in the `GitEventsMonitor`? It looks more like the latter. If so, it would have been better create a separate bug for it and handle it in a separated PR and not mix unrelated changes.

Reply:
Because when we change it in Git it does not throw a changed event. It creates a new file, deletes the old one and renames the new one to "HEAD"

Reply:
Do you think that is worth documenting in the code?